civis/futures.py

```diff
@@ -286,6 +286,53 @@ def __init__(self, job_id, run_id,
                          client=client,
                          poll_on_creation=poll_on_creation)
         self._max_n_retries = max_n_retries
+        self._exception_handled = False
+        self.add_done_callback(self._set_job_exception)
+
+    @staticmethod
+    def _set_job_exception(fut):
```

Comment:
It's worth documenting that this method is intended for use as a callback. You could add a docstring with that mention.